### PR TITLE
WIP adding @canEdit to card format components

### PR DESCRIPTION
**Pull request by lukemelia**
- [ ] Stack context
  - [x] contains
  - [x] containsMany
  - [x] linksTo
  - [x] linksToMany
- [ ] Preview panel
- [ ] Guest mode

TODO: Make sure all fields that we defined (every place where FieldDef is used) respect the `@canEdit`

### CODE REVIEW

File: packages/host/app/resources/realm-session.ts [hunk 1]
```diff
@@ -104,11 +107,11 @@ export class RealmSessionResource extends Resource<Args> {
   }
 
   get canRead() {
-    return this.token?.permissions?.includes('read');
+    return !!this.token?.permissions?.includes('read');
```
Comment (lukemelia) [inline packages/host/app/resources/realm-session.ts:110]: might want to consider whether realm is publicly readable here

**Comment by jurgenwerk** [comment]
Github is acting really strangely, it's been stuck in "Processing updates" since yesterday and no new commits are shown even though they are pushed and can be seen in https://github.com/cardstack/boxel/commits/cs-6374-add-canedit-arg-to-basedef-and-make-boxel-set-this-on-cards/

image

**Comment by jurgenwerk** [comment]
@lukemelia I created a copy of this PR (https://github.com/cardstack/boxel/pull/1237) because this one is broken, it doesn't process new commits 🤷‍♂️ 😵‍💫

**Comment by lukemelia** [comment]
Weird!

**Pull request closed without merging**